### ISSUE TITLE
pyright-langserver reports unused variables even when supressed by config

### ISSUE THREAD

**Issue by herberthamaral** [microsoft/pyright#2307]
**Describe the bug**

pyright-langserver reports non-accessed variables even when suppressed by config. This doesn't happen if I call pyright directly.

**To Reproduce**

Example source code (`example.py`):
```python
def example(*args):
    pass
```
It shouldn't report any issues. In fact, if I run `pyright` on the above file I get the following output:

```
$ pyright src/sales/views/sales_advertisement_detail_views/example.py 
No configuration file found.
No pyproject.toml file found.
Assuming Python platform Linux
Searching for source files
Found 1 source file
0 errors, 0 warnings, 0 infos 
Completed in 0.534sec
```

However, if I open the very same file in my editor (doom-emacs, emacs 27.2 running on linux) and run `strace -p $lspid -s9999 -e trace=read,write` on `pyright-langserver`'s process, I see the following call (redacted for clarity):

```
read(0, "Content-Length: 454\r\n\r\n{\"jsonrpc\":\"2.0\",\"method\":\"textDocument/codeAction\",\"params\":{\"textDocument\":{\"uri\":\"file:///tmp/testproject/example.py\"},\"range\":{\"start\":{\"line\":0,\"character\":0},\"end\":{\"line\":0,\"character\":0}},\"context\":{\"diagnostics\":[{\"range\":{\"start\":{\"line\":0,\"character\":13},\"end\":{\"line\":0,\"character\":17}},\"message\":\"\\\"args\\\" is not accessed\",\"severity\":4,\"source\":\"Pyright\",\"tags\":[1]}]}},\"id\":39364}\n", 65536) = 477
```
The same thing happens when I use a `pyrightconfig.json` file.

**Expected behavior**

`pyright-langserver` should follow `pyright`'s behavior on reporting errors.

**Screenshots or Code**

Python code:

```python
def example(*args):
    pass
```

`pyrightconfig.json` example:

```
{
    "reportUnusedClass": false,
    "reportUnusedFunction": false,
    "reportUnusedVariable": false,
}
```

**VS Code extension or command-line**
 pyright 1.1.162

**Additional context**

I don't know how to better reproduce this issue. Please let me know if there's a better way to reproduce issues related to `pyright-langserver`.

**Comment by erictraut** [comment]
Pyright reports hints to tell an editor that it should display certain text as "dimmed" to indicate to that the symbol is unaccessed. These hints are documented in the LSP standard as "hint" diagnostics (severity 4) and an explicit "tag" of 1 to indicate that they should be displayed as dimmed. You can see the severity and tags fields in the output you pasted above. These hints are returned only if the client requests them, so clearly your client is requesting them. It's up to the client to display these hints appropriately. Some clients display these hints as regular diagnostics, but that is not appropriate. If your editor is doing this, please report it as a bug to the maintainer of the editor or the LSP.

Pyright will never emit a real diagnostic for an unused parameter.
